packages/compass-global-writes/src/components/create-shard-key-form.tsx

```diff
@@ -0,0 +1,339 @@
+                    Learn more
+                  </Link>
+                </Body>
+              </div>
+            </Radio>
+            <Radio
+              id="hashed-index"
+              value="hashed-index"
+              checked={selectedAdvancedOption === 'hashed-index'}
+            >
+              <div>
+                <Label htmlFor="hashed-index">
+                  Use hashed index as the shard key
+                </Label>
+                <Body>
+                  Improve even distribution of the sharded data by hashing the
+                  second field of the shard key.{' '}
+                  <Link href="https://docs.atlas.mongodb.com/data-explorer/global-writes/#optional-expand-advanced-shard-key-configuration-section-to-specify-how-to-shard-the-collection">
+                    Learn more
+                  </Link>
+                </Body>
+              </div>
+            </Radio>
+          </RadioGroup>
+          {selectedAdvancedOption === 'hashed-index' && (
+            <div className={cx(contentStyles, hasedIndexOptionsStyles)}>
+              <Checkbox
+                data-testid="presplit-data-checkbox"
+                onChange={() => setIsPreSplitData(!isPreSplitData)}
+                label="Pre-split data for even distribution."
+                checked={isPreSplitData}
+              />
+              <div className={chunksInputStyles}>
+                <TextInput
+                  data-testid="chunks-per-shard-input"
+                  id="chunks-per-shard"
+                  aria-labelledby="Chunks per shard"
+                  disabled={!isPreSplitData}
+                  type="number"
+                  placeholder="Chunks"
+                  min={0}
+                  value={numInitialChunks}
+                  onChange={(event) => setNumInitialChunks(event.target.value)}
+                />
+                <Body>chunks per shard.</Body>
+              </div>
+            </div>
+          )}
+        </Accordion>
+        <div>
+          <Button
+            data-testid="shard-collection-button"
+            onClick={onSubmit}
+            disabled={
+              !secondShardKey || isSubmittingForSharding || isCancellingSharding
+            }
+            variant="primary"
+            isLoading={isSubmittingForSharding}
+          >
+            Shard Collection
+          </Button>
+        </div>
+      </div>
+    </>
+  );
+}
+
+export default connect(
+  (state: RootState) => {
+    return {
+      namespace: state.namespace,
+      isSubmittingForSharding: [
+        ShardingStatuses.SUBMITTING_FOR_SHARDING,
+        ShardingStatuses.SUBMITTING_FOR_SHARDING_ERROR,
+      ].includes(state.status),
+      isCancellingSharding: [
+        ShardingStatuses.CANCELLING_SHARDING,
+        ShardingStatuses.CANCELLING_SHARDING_ERROR,
+      ].includes(state.status),
+    };
+  },
+  {
+    onCreateShardKey: createShardKey,
+  }
+)(CreateShardKeyForm);
```

Comment:
Another small change - this is now connected directly, and it works with error/no error statuses.